### PR TITLE
added helm chart

### PR DESCRIPTION
**Pull request by haarchri**
feature request completed for #60 
please have a look ;)

### CODE REVIEW

File: deploy/helm/kube-cleanup-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+{{ with .Values.podLabels }}{{ toYaml . | indent 8 }}{{ end }}
+      annotations:
+{{ with .Values.podAnnotations }}{{ toYaml . | indent 8 }}{{ end }}
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: {{ .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- with .Values.args }}
+          args: {{ toYaml . | nindent 16 }}
```
Comment (lwolf) [inline deploy/helm/kube-cleanup-operator/templates/deployment.yaml:43]: Here and below, `nindent 12` looks better than`nindent 16` when rendered

File: deploy/helm/kube-cleanup-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+* <https://github.com/lwolf/kube-cleanup-operator.git>
+
+## Values
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| affinity | object | `{}` |  |
+| annotations | object | `{}` |  |
+| args[0] | string | `"--namespace=default"` |  |
```
Comment (lwolf) [inline deploy/helm/kube-cleanup-operator/README.md:17]: List of args could change at any time, and it doesn't require specific order, so may be change this to something like:
```
args | array | list of cli arguments for the binary, like `--namespace=default`, consult README for the list of arguments
```

File: deploy/helm/kube-cleanup-operator/values.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  # If not set and create is true, a name is generated using the fullname template
+  name:
+
+rbac:
+  # Specifies whether RBAC resources should be created
+  create: true
+
+## Arguments for the exporter binary
+##
+args:
```
Comment (lwolf) [inline deploy/helm/kube-cleanup-operator/values.yaml:93]: Let's not have default values here, it's extremely hard to change defaults once set. 
```
 # - --namespace=default
 # - --delete-successful-after=5m
 # - --delete-failed-after=120m
 # - --delete-pending-pods-after=60m
 # - --delete-evicted-pods-after=60m
 # - --delete-orphaned-pods-after=60m
 # - --legacy-mode=false
args: []

```

**Review by lwolf — CHANGES_REQUESTED** [review]
Thanks for the PR, I left a couple of minor change requests.

It looks like you generated this chart using old helm2 which is in process of being retired (https://github.com/helm/helm/issues/8343).
Do you mind generating it using helm3?

File: deploy/helm/kube-cleanup-operator/values.yaml (hunk 3 above)
Comment (haarchri) [reply deploy/helm/kube-cleanup-operator/values.yaml]: done

File: deploy/helm/kube-cleanup-operator/README.md (hunk 2 above)
Comment (haarchri) [reply deploy/helm/kube-cleanup-operator/README.md:17]: done

File: deploy/helm/kube-cleanup-operator/templates/deployment.yaml (hunk 1 above)
Comment (haarchri) [reply deploy/helm/kube-cleanup-operator/templates/deployment.yaml]: done

File: deploy/helm/kube-cleanup-operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  create: true
+  # The name of the ServiceAccount to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
+
+rbac:
+  # Specifies whether RBAC resources should be created
+  create: true
+
+## Arguments for the exporter binary
```
[the remark this answers is not in the document]
Comment (haarchri) [reply deploy/helm/kube-cleanup-operator/values.yaml]: done

**Comment by haarchri** [comment]
`It looks like you generated this chart using old helm2 which is in process of being retired `

it is running with helm3 
`version.BuildInfo{Version:"v3.4.2"`

what is in your opinion not v3 ?

**Comment by lwolf** [comment]
At least the value  of `apiVersion: v1` signaling that it was generated for helm2.
Also, helm3 change the way how image.tag is populated, it now uses `appVersion` field for the tag, which allows one to not set `tag` to latest

Here is the output from helm2 vs helm3 of the Chart.yaml file:
```
$ helm3 create deploy/helm/test3
$ cat deploy/helm/test3/Chart.yaml

apiVersion: v2
name: test3
description: A Helm chart for Kubernetes

# A chart can be either an 'application' or a 'library' chart.
#
# Application charts are a collection of templates that can be packaged into versioned archives
# to be deployed.
#
# Library charts provide useful utilities or functions for the chart developer. They're included as
# a dependency of application charts to inject those utilities and functions into the rendering
# pipeline. Library charts do not define any templates and therefore cannot be deployed.
type: application

# This is the chart version. This version number should be incremented each time you make changes
# to the chart and its templates, including the app version.
# Versions are expected to follow Semantic Versioning (https://semver.org/)
version: 0.1.0

# This is the version number of the application being deployed. This version number should be
# incremented each time you make changes to the application. Versions are not expected to
# follow Semantic Versioning. They should reflect the version the application is using.
appVersion: 1.16.0

```

vs
```
$ helm create deploy/helm/test
$ cat deploy/helm/test/Chart.yaml

apiVersion: v1
appVersion: "1.0"
description: A Helm chart for Kubernetes
name: test
version: 0.1.0

```

**Comment by haarchri** [comment]
adopted Charts.yaml for helmv3 - in deployment the field was present ;)

any other change requests ?

**Pull request merged**